pallets/dca/src/lib.rs

```diff
@@ -1092,16 +1139,36 @@ impl<T: Config> Pallet<T> {
 	) -> Result<Balance, DispatchError> {
 		let amount = if asset_id == T::NativeAssetId::get() {
 			asset_amount
-		} else {
+		} else if T::SwappablePaymentAssetSupport::is_transaction_fee_currency(asset_id) {
 			let price = T::NativePriceOracle::price(asset_id).ok_or(Error::<T>::CalculatingPriceError)?;
 
 			multiply_by_rational_with_rounding(asset_amount, price.n, price.d, Rounding::Up)
 				.ok_or(ArithmeticError::Overflow)?
+		} else {
+			let fee_amount_in_dot = Self::convert_to_polkadot_native_asset(asset_amount)?;
```

Comment:
this does not seem right. 

it tries to convert asset_amount to DOT, by calling convert_to_polkadot.  and that functions uses hdx/dot price. but asset amount here is not HDX. 
or am i missing something ?

Reply:
maybe the variable naming could be improved, so `asset_amount` could be renamed to `native_amount`

The name of the function `convert_native_amount_to_currency`, so this function is supposed to be used for converting native amount (so HDX amount) to some currency.

Reply:
ah ok. i haven't noticed that it already converts native amount